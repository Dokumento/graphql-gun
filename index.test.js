/* global describe, it, expect */
const gql = require('graphql-tag');
const Gun = require('gun');
const graphqlGun = require('./');

const gun = Gun();

describe("graphqlGun", () => {
  it("can get things down the gun chain", async () => {
    gun.get('foo').put({bar: 'baz'});
    gun.get('foo').get('bar').put({hello: 'world'});
        
    const results = await graphqlGun(gql`{
      foo {
        bar {
          hello
        }
      }
    }`, gun);
    
    expect(results.foo.bar.hello).toEqual('baz');
    expect(results).toMatchSnapshot();
  })
  
  it("lets you grab the chain at any point", async () => {
    gun.get('foo').put({bar: 'pop'});
    
    const results = await graphqlGun(gql`{
      foo {
        bar {
          _chain
          hello
        }
      }
    }`, gun);
    
    expect(results).toMatchSnapshot();
    
    await (new Promise((resolve) => {
      results.foo.bar._chain.on((value, key) => {
        expect(key).toEqual('bar')
        expect(value).toEqual('pop')
        resolve()
      }, {changed: true})
    
      gun.get('foo').get('bar').put({some: 'stuff'});
    }));
    
  })
});