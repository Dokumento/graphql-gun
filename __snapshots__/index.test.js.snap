// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphqlGun can get things down the gun chain 1`] = `
Object {
  "foo": Object {
    "bar": Object {
      "hello": "baz",
    },
  },
}
`;

exports[`graphqlGun lets you grab the chain at any point 1`] = `
Object {
  "foo": Object {
    "bar": Object {
      "_chain": undefined,
      "hello": "pop",
    },
  },
}
`;
